roachtest: add more exceptions to bank/zerosum-restart

tbg opened the pull request:
Fixes #33683.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33707)
<!-- Reviewable:end -->

petermattis reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33707)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/bank.go, line 348 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/33707#-LWBek475tt6BuF0rvcg:-LWBek475tt6BuF0rvch:b-3kso45) ([raw file](https://github.com/cockroachdb/cockroach/blob/29f70ccb9eb3b14d633d8e89d082a83ec403791b/pkg/cmd/roachtest/bank.go#L348)):*
> ```Go
> 
> func isExpectedRelocateError(err error) bool {
> 	return testutils.IsError(err, "(descriptor changed|unable to remove replica .* which is not present|unable to add replica .* which is already present|received invalid ChangeReplicasTrigger .* to remove self|breaker open|cannot up-replicate|error looking up store)")
> ```

This is getting a bit excessive. When I originally added these, there were explanations for why the various errors were occurring. Do you know why/how `error looking up store` is occurring? That one seems particularly suspicious. 


<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33707)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/bank.go, line 348 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/33707#-LWBek475tt6BuF0rvcg:-LWCkNAa4D2sdeJWDUhD:b-x4pz4t) ([raw file](https://github.com/cockroachdb/cockroach/blob/29f70ccb9eb3b14d633d8e89d082a83ec403791b/pkg/cmd/roachtest/bank.go#L348)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

This is getting a bit excessive. When I originally added these, there were explanations for why the various errors were occurring. Do you know why/how `error looking up store` is occurring? That one seems particularly suspicious. 
</blockquote></details>

yeah... Is this the new `testing.RetriableErr()` of ill repute?


<!-- Sent from Reviewable.io -->

tbg commented:
They're not suspicious if you ask me:

https://github.com/cockroachdb/cockroach/blob/2558dccbcda09c0eae28f816a05adb8620155175/pkg/sql/relocate.go#L171-L183

https://github.com/cockroachdb/cockroach/blob/bd00a26afd75b84160735ad848a7d48e3cc8a660/pkg/storage/replica_command.go#L1232-L1239

I agree that it's unsavory. These errors can also happen to users. Ultimately they're the result on relying on an asynchronous mechanism (Gossip). To adress them, I think, we ultimately need to either a) poll Gossip for a bit before declaring that something "isn't there" or b) have fallbacks to the KV store for everything (again with a timeout, to make this more reliable).

I won't be working on that anytime soon though, so my question is whether you prefer this to merge as-is (I'm happy to backfill the error descriptions with code snippets) or whether we'll want to skip the test again.

petermattis reviewed (commented):
> I won't be working on that anytime soon though, so my question is whether you prefer this to merge as-is (I'm happy to backfill the error descriptions with code snippets) or whether we'll want to skip the test again.

I'd prefer to leave the test as skipped until someone has time to more thoroughly investigate and fix. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33707#-:-LWIGcmi-3lhTcm8h6Ag:b-a8or2c)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg commented:
> I'd prefer to leave the test as skipped until someone has time to more thoroughly investigate and fix.

👍  done, changed the PR accordingly and also referenced the known failure modes I know of.

bors r=petermattis

tbg committed:
roachtest: skip bank/zerosum-restart

It flakes on various relocate errors that are "expected" when nodes
are freshly restarted due to Gossip asynchronicity. However we've
decided to work on that root cause instead of adding more random
exceptions to the test. As a result, we're skipping it for now.

Release note: None

tbg commented:
Had removed a bracket too many.

tbg commented:
bors r=petermattis

tbg commented:
TestLint flake.

bors r=petermattis

andreimatei commented on pkg/cmd/roachtest/acceptance.go:
```diff
@@ -33,7 +37,7 @@ func registerAcceptance(r *registry) {
 		{"bank/node-restart", runBankNodeRestart},
 		{"bank/zerosum-splits", runBankNodeZeroSum},
-		{"bank/zerosum-restart", runBankZeroSumRestart},
+		// {"bank/zerosum-restart", runBankZeroSumRestart},
```
wait - why comment this out instead of putting a `t.Skip()` at the beginning of `runBankZeroSumRestart()` ?
